### PR TITLE
Fixed #3535 - Variable clash ($request vs $_REQUEST) in modules/Emails/Email.php email2Send() 

### LINKED ISSUE #3535: Variable clash ($request vs $_REQUEST) in modules/Emails/Email.php email2Send()
#### Issue
$_REQUEST is already passed to the function as $request.
see modules/Emails/EmailUIAjax.php case 'sendEmail'.

#### Expected Behaviour
In general, we should avoid using global variables in classes.

#### Actual Behaviour
Both variables are in use.

#### Possible Fix
In this function, we should just use $request.

#### Your Environment
* SuiteCRM Version used: 7.8.3
* Browser name and version (e.g. Chrome 58 (64-bit)):
* Environment name and version (e.g. MySQL 5.7, PHP 7.0):
* Operating System and version (e.g Ubuntu 16.04):

### PR DESCRIPTION
**Pull request by Dillon-Brown**
## Description

Replaced uses of $_REQUEST within the email2Send() function as it is already passed into the function as $request.

## Motivation and Context

This fix removes unnecessary globals within the email2Send class.

Issue reference: #3535

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [X] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [X] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: modules/Emails/Email.php [hunk 1]
```diff
@@ -437,393 +438,407 @@ function isDraftEmail($request)
...
-	function email2Send($request) {
-		global $mod_strings;
-		global $app_strings;
-		global $current_user;
-		global $sugar_config;
-		global $locale;
-		global $timedate;
-		global $beanList;
-		global $beanFiles;
+    function email2Send($request)
```
Comment (daniel-samson) [inline modules/Emails/Email.php]: Please update PHP Docs.

File: modules/Emails/Email.php [hunk 2]
```diff
@@ -437,393 +438,407 @@ function isDraftEmail($request)
...
+                    $fileGUID = preg_replace('/[^a-z0-9\-]/', "", substr($file, 0, 36));
+                    $fileLocation = $this->et->userCacheDir . "/{$fileGUID}";
+                    $filename = substr($file, 36,
+                        strlen($file)); // strip GUID	for PHPMailer class to name outbound file
+
+                    $mail->AddAttachment($fileLocation, $filename, 'base64', $this->email2GetMime($fileLocation));
+                    //$mail->AddAttachment($fileLocation, $filename, 'base64');
+
+                    // only save attachments if we're archiving or drafting
+                    if ((($this->type == 'draft') && !empty($this->id)) || (isset($request['saveToSugar']) && $request['saveToSugar'] == 1)) {
```
Comment (daniel-samson) [inline modules/Emails/Email.php]: please replace == with ===

File: modules/Emails/Email.php [hunk 3]
```diff
@@ -437,393 +438,407 @@ function isDraftEmail($request)
...
+
+                    $filename = $docRev->filename;
+                    $docGUID = preg_replace('/[^a-z0-9\-]/', "", $docRev->id);
+                    $fileLocation = "upload://{$docGUID}";
+                    $mime_type = $docRev->file_mime_type;
+                    $mail->AddAttachment($fileLocation,
+                        $locale->translateCharsetMIME(trim($filename), 'UTF-8', $OBCharset), 'base64', $mime_type);
+
+                    // only save attachments if we're archiving or drafting
+                    if ((($this->type == 'draft') && !empty($this->id)) || (isset($request['saveToSugar']) && $request['saveToSugar'] == 1)) {
```
Comment (daniel-samson) [inline modules/Emails/Email.php]: please change == with ===

File: modules/Emails/Email.php [hunk 4]
```diff
@@ -437,393 +438,407 @@ function isDraftEmail($request)
...
+                    $note->retrieve($noteId);
+                    if (!empty($note->id)) {
+                        $filename = $note->filename;
+                        $noteGUID = preg_replace('/[^a-z0-9\-]/', "", $note->id);
+                        $fileLocation = "upload://{$noteGUID}";
+                        $mime_type = $note->file_mime_type;
+                        if (!$note->embed_flag) {
+                            $mail->AddAttachment($fileLocation, $filename, 'base64', $mime_type);
+                            // only save attachments if we're archiving or drafting
+                            if ((($this->type == 'draft') && !empty($this->id)) || (isset($request['saveToSugar']) && $request['saveToSugar'] == 1)) {
```
Comment (daniel-samson) [inline modules/Emails/Email.php]: replace == with ===

File: modules/Emails/Email.php [hunk 5]
```diff
@@ -437,393 +438,407 @@ function isDraftEmail($request)
...
+
+                                if ($note->parent_id != $this->id) {
```
Comment (daniel-samson) [inline modules/Emails/Email.php]: replace != with !==

File: modules/Emails/Email.php [hunk 6]
```diff
@@ -437,393 +438,407 @@ function isDraftEmail($request)
...
-			$this->status = 'draft';
-			$forceSave = true;
-		}
+        /**********************************************************************
+         * Final Touches
+         */
+        /* save email to sugar? */
+        $forceSave = false;
+
+        if ($this->type == 'draft' && !isset($request['saveDraft'])) {
```
Comment (daniel-samson) [inline modules/Emails/Email.php]: replace == with ===

File: modules/Emails/Email.php [hunk 7]
```diff
@@ -834,128 +849,130 @@ function email2Send($request) {
...
+            (($request['composeType'] == 'reply') || ($request['composeType'] == 'replyAll') || ($request['composeType'] == 'replyCase')) && ($orignialId != $this->id)
+        ) {
+            $originalEmail = new Email();
+            $originalEmail->retrieve($orignialId);
+            $originalEmail->reply_to_status = 1;
+            $originalEmail->save();
+            $this->reply_to_status = 0;
+        } // if
+
+        if ($request['composeType'] == 'reply' || $request['composeType'] == 'replyCase') {
```
Comment (daniel-samson) [inline modules/Emails/Email.php]: replace == with ===

File: modules/Emails/Email.php [hunk 8]
```diff
@@ -834,128 +849,130 @@ function email2Send($request) {
...
+                    }
+                }
+            }
+        }
 
-						} // if
 
-					} // if
+        if ($forceSave ||
+            $this->type == 'draft' ||
```
Comment (daniel-samson) [inline modules/Emails/Email.php]: replace == with ===

File: modules/Emails/Email.php [hunk 9]
```diff
@@ -834,128 +849,130 @@ function email2Send($request) {
...
-					} // else
-				} else {
-					$GLOBALS['log']->debug("could not copy email to {$ie->mailbox} sent folder as its empty");
-				} // else
-			} // if
-		} // if
-		return true;
-	} // end email2send
+        if (!empty($request['fromAccount'])) {
+            if (isset($ie->id) && !$ie->isPop3Protocol() && $mail->oe->mail_smtptype != 'gmail') {
```
Comment (daniel-samson) [inline modules/Emails/Email.php]: replace != with !==

**Comment by Dillon-Brown** [comment]
Superseded by #4343

**Pull request closed without merging**